test.js

```diff
@@ -83,3 +83,16 @@ test('works with circular dependencies', t => {
 	t.is(childrenCalls, 4);
 	clearModule.all();
 });
+
+test('clear with filter', t => {
+	const id = './fixture-filter';
+	require(id);
+
+	const parentmodule = Object.keys(require.cache).find(v => v.match(/parent-module/));
```

Comment:
`.match()` is not the correct method to use for this. You want to use `.test()`, but in this case, just `.includes()` would be better.

Also use descriptive variable names.